File: src/main/java/es/us/dp1/l4_01_24_25/upstream/statistic/Achievement.java
package es.us.dp1.l4_01_24_25.upstream.statistic;

import es.us.dp1.l4_01_24_25.upstream.model.NamedEntity;
import jakarta.persistence.Entity;
import jakarta.persistence.EnumType;
import jakarta.persistence.Enumerated;
import jakarta.validation.constraints.Min;
import jakarta.validation.constraints.NotNull;
import lombok.Getter;
import lombok.Setter;

@Entity
@Getter
@Setter
public class Achievement extends NamedEntity {
    
    private String description;
    
    private String badgeImage;
    
    @Min(0)
    private Double threshold;

    @Enumerated(EnumType.STRING)
    @NotNull
    Metric metric;

}

File: frontend/src/interfaces/Achievement.ts
export default interface Achievement {
  id: number;
  name: string;
  description: string;
  badgeImage: string;
  threshold: number;
  metric: string;
  actualDescription?: string;
}

export const emptyAchievement: Achievement = {id: 0, name: '', description: '', badgeImage: '',
    threshold: 1, metric: 'GAMES_PLAYED', actualDescription: '',};